django_password_validators/password_history/password_validation.py

```diff
@@ -42,18 +60,41 @@ def validate(self, password, user=None):
 
         if not self._user_ok(user):
             return
-
         for user_config in UserPasswordHistoryConfig.objects.filter(user=user):
             password_hash = user_config.make_password_hash(password)
             try:
-                PasswordHistory.objects.get(
-                    user_config=user_config,
-                    password=password_hash
-                )
-                raise ValidationError(
-                    _("You can not use a password that is already used in this application."),
-                    code='password_used'
-                )
+                if self.lookup_range == 0:
+                    raise PasswordHistory.DoesNotExist
+
+                current_user_passwords = PasswordHistory.objects.filter(
+                    user_config=user_config
+                ).order_by('date')
+
+                num_user_passwords = len(current_user_passwords)
+
+                if self.lookup_range >= num_user_passwords:
+                    PasswordHistory.objects.get(
+                        user_config=user_config,
+                        password=password_hash
+                    )
+                    raise self.validation_error
+
+                else:
+                    # At this point, there are passwords we have to delete
+                    for entry in current_user_passwords[
```

Comment:
@spokerman12, it would make the validation logic and tests simpler to move the deletion to the `password_changed()` method after the password is saved. Then the validation code can just focus on validation and all the database changes are actually in one method. What do you think?